test/Integration/SqlOutputBindingIntegrationTests.cs

```diff
@@ -472,5 +472,34 @@ public void AddProductWithDifferentPropertiesTest(SupportedLanguages lang)
             // Verify result
             Assert.Equal("test2", this.ExecuteScalar($"select Name from Products where ProductId=0"));
         }
+
+        /// <summary>
+        /// Tests that when upserting an item with no properties, an error is thrown.
+        /// </summary>
+        [Theory]
+        [SqlInlineData()]
+        // Only the JavaScript function passes an empty JSON to the SQL extension.
+        // C#, Java, and Python throw an error while creating the Product object in the function and in PowerShell,
```

Comment:
Yes, for example, in PowerShell: `Cannot insert the value NULL into column 'ProductId', table 'MyDatabase.dbo.Products'; column does not allow nulls. UPDATE fails.`.

Reply:
What's the error in the others?

Reply:
The other language functions throw an error for having a null value for the parameters while creating the Product object.
CSharp:
```csharp
[2023-02-03T21:10:15.602Z] Executed 'AddProductParams' (Failed, Id=0ce8be95-9c9e-426d-a2cd-9fead8408b2e, Duration=398ms)
[2023-02-03T21:10:15.602Z] System.Private.CoreLib: Exception while executing function: AddProductParams. System.Private.CoreLib: Value cannot be null. (Parameter 's').
``` 
Python:
```python
[2023-02-03T21:08:56.639Z] Executed 'Functions.AddProductParams' (Failed, Id=c1d989d7-60eb-426c-b33f-676d66d4df75, Duration=148ms)
[2023-02-03T21:08:56.639Z] System.Private.CoreLib: Exception while executing function: Functions.AddProductParams. System.Private.CoreLib: Result: Failure
[2023-02-03T21:08:56.639Z] Exception: KeyError: 'productId'
[2023-02-03T21:08:56.639Z] Stack:   File "/Users/lucyzhang/.nvm/versions/node/v14.18.2/lib/node_modules/azure-functions-core-tools/bin/workers/python/3.8/OSX/X64/azure_functions_worker/dispatcher.py", line 458, in _handle__invocation_request
[2023-02-03T21:08:56.639Z]     call_result = await self._loop.run_in_executor(
[2023-02-03T21:08:56.639Z]   File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/concurrent/futures/thread.py", line 57, in run
[2023-02-03T21:08:56.639Z]     result = self.fn(*self.args, **self.kwargs)
[2023-02-03T21:08:56.639Z]   File "/Users/lucyzhang/.nvm/versions/node/v14.18.2/lib/node_modules/azure-functions-core-tools/bin/workers/python/3.8/OSX/X64/azure_functions_worker/dispatcher.py", line 701, in _run_sync_func
[2023-02-03T21:08:56.639Z]     return ExtensionManager.get_sync_invocation_wrapper(context,
[2023-02-03T21:08:56.639Z]   File "/Users/lucyzhang/.nvm/versions/node/v14.18.2/lib/node_modules/azure-functions-core-tools/bin/workers/python/3.8/OSX/X64/azure_functions_worker/extension.py", line 215, in _raw_invocation_wrapper
[2023-02-03T21:08:56.639Z]     result = function(**args)
[2023-02-03T21:08:56.639Z]   File "/Users/lucyzhang/GitProjects/azure-functions-sql-extension/samples/samples-python/AddProductParams/__init__.py", line 12, in main
[2023-02-03T21:08:56.639Z]     row = func.SqlRow(Product(params["productId"][0], params["name"][0], params["cost"][0]))
[2023-02-03T21:08:56.639Z] .
```
Java:
```java
[2023-02-03T21:07:47.873Z] Executed 'Functions.AddProductParams' (Failed, Id=cf99e269-0285-42b9-a564-58d19e1e50c2, Duration=213ms)
[2023-02-03T21:07:47.873Z] System.Private.CoreLib: Exception while executing function: Functions.AddProductParams. System.Private.CoreLib: Result: Failure
[2023-02-03T21:07:47.873Z] Exception: NumberFormatException: null
[2023-02-03T21:07:47.873Z] Stack: java.lang.reflect.InvocationTargetException
[2023-02-03T21:07:47.873Z]      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2023-02-03T21:07:47.873Z]      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2023-02-03T21:07:47.873Z]      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2023-02-03T21:07:47.873Z]      at java.lang.reflect.Method.invoke(Method.java:498)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.broker.JavaMethodInvokeInfo.invoke(JavaMethodInvokeInfo.java:22)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.broker.JavaMethodExecutorImpl.execute(JavaMethodExecutorImpl.java:22)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.chain.FunctionExecutionMiddleware.invoke(FunctionExecutionMiddleware.java:19)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.chain.InvocationChain.doNext(InvocationChain.java:21)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.invokeMethod(JavaFunctionBroker.java:80)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:37)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:12)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.handler.MessageHandler.handle(MessageHandler.java:44)
[2023-02-03T21:07:47.873Z]      at com.microsoft.azure.functions.worker.JavaWorkerClient$StreamingMessagePeer.lambda$onNext$0(JavaWorkerClient.java:93)
[2023-02-03T21:07:47.873Z]      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[2023-02-03T21:07:47.873Z]      at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2023-02-03T21:07:47.873Z]      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2023-02-03T21:07:47.873Z]      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2023-02-03T21:07:47.874Z]      at java.lang.Thread.run(Thread.java:750)
[2023-02-03T21:07:47.874Z] Caused by: java.lang.NumberFormatException: null
[2023-02-03T21:07:47.874Z]      at java.lang.Integer.parseInt(Integer.java:542)
[2023-02-03T21:07:47.874Z]      at java.lang.Integer.parseInt(Integer.java:615)
[2023-02-03T21:07:47.874Z]      at com.function.AddProductParams.run(AddProductParams.java:38)
[2023-02-03T21:07:47.874Z]      ... 18 more
```

Reply:
Hm, ok well that's not my favorite error since I don't think it's very useful for helping the customer find out what was wrong (basically that they were missing columns). Can we do anything to make that better? Even if it's just adding a suggestion of what to look at when the error is thrown could help avoid some confusion.

Reply:
Hm do you mean adding a suggestion to our docs? Maybe we can have a troubleshooting section?

Reply:
We should, but no in this case it would be best if we could make the actual exception include at least a hint to what the issue might be. We should be able to detect when "invalid" column values are used, right? For example all primary keys must be non-null.

Reply:
For C#, Java, Python, the functions are throwing the error while creating the Product object before anything is sent to the sql extension so we wouldn't be able to modify the exception.

Reply:
Ok, for anything that is outside our extension then let's definitely get it documented

@VasuBhog Stuff like this will be good to capture for your TSG work

Reply:
Yup been following this. I will be sure to add this to potential issues for each of the languages that users may end up having. Then create appropriate TSGs for them.